### ISSUE TITLE
possiable improved tests for exercise/conversions/from_into.rs

### ISSUE THREAD

**Issue by emiddleton** [rust-lang/rustlings#358]
I am working through the questions and I noticed that what seems to be the desired answer has some issues with various malformed string types.

The error cases I was thinking of were 
* missing age eg. "Mark,"
* missing comma and age eg. "Mark"
* missing name and age eg. ","
* missing name eg. . ",1"
* missing name and invalid age eg. ",one".

To handle these without panicking I ended up with
 
```rust
impl From<&str> for Person {
    fn from(s: &str) -> Person {
        if s.len() == 0 {
            Person::default()
        } else {
            let mut person_itr = s.split(',');
            match person_itr.next().map(|name| {
                (
                    if name.is_empty() {
                        Err("Missing Name")
                    } else {
                        Ok(name.to_string())
                    },
                    person_itr.collect::<String>().parse::<usize>(),
                )
            }) {
                Some((Ok(name), Ok(age))) => Person { name, age },
                _ => Person::default(),
            }
        }
    }
}
```

Without the above test cases I would have assumed the various elements were present in the string and thrown an exception if missing.

**Comment by AbdouSeck** [comment]
Hi @emiddleton,

Thanks for bringing this up. I am assuming that the panicking is due to unclear instructions.
For the error cases you're thinking of, if we use the instructions, you should get `Person {name: "John", age: 30}`. The only exception may be `",1"`. With this one, you should also get `Person {name: "John", age: 30}`, but the instructions are not quite clear about what to do when only the name part of the string is empty.

**Next steps**:
1. Add the clarification that a name cannot be an empty string, so `Person::default()` should be returned.
2. Add more test cases using the error cases you've provided here.


If you'd like, please feel free to add those changes.

Please let me know what you think.

Thank you!

Abdou

**Comment by AbdouSeck** [comment]
@emiddleton I believe this was addressed in #368. Please feel free to reach out if something looks amiss. Otherwise, @fmoko can go ahead and close this issue.

Thank you!

Abdou
